server/monitor/src/test/java/org/apache/accumulo/monitor/util/AccumuloMonitorAppenderTest.java

```diff
@@ -174,11 +175,15 @@ private static boolean verifyAppender(AccumuloMonitorAppender parent, String new
     if (childAppenders == null || !childAppenders.hasMoreElements()) {
       return false;
     }
-    AppenderSkeleton child = (AppenderSkeleton) childAppenders.nextElement();
-    if (childAppenders.hasMoreElements()) {
-      Assert.fail("Appender should never have more than one child");
+    try {
+      AppenderSkeleton child = (AppenderSkeleton) childAppenders.nextElement();
+      if (childAppenders.hasMoreElements()) {
+        Assert.fail("Appender should never have more than one child");
+      }
+      return ("Appender for " + newLocName).equals(child.getName());
+    } catch (NoSuchElementException e) {
```

Comment:
Is this an unrelated bug that we need to fix or is this a Hadoop3-specific thing?

Reply:
I'm not positive, I think it's just a race condition. It failed on me once -- didn't dig into it deeply.

Reply:
Sounds like the thing you mentioned on https://issues.apache.org/jira/browse/ACCUMULO-4409
Just checking that it's not something new. :smile:

Reply:
Also, please don't include that if you commit. It will mask a potentially more serious problem with the log appender, and if it happens again, I'd prefer the test fail so I can troubleshoot it further.

Reply:
Rofl, no recollection at all seeing that before. Will revert.